framework/include/bcs/DirichletBC.h

```diff
@@ -36,7 +36,7 @@ class DirichletBC : public NodalBC
   virtual Real computeQpResidual();
 
   /// The value for this BC
-  const Real & _value;
+  const PostprocessorValue & _value;
```

Comment:
I'll just copy my comment from your other PR:

I don't like this for one of our most basic classes in MOOSE. This class is used (and inherited from) by EVERYONE... and now it's going to show that the value is expecting a PostprocessorName? And now Peacock is going to suggest the value be an Postprocessor in your input file?

The right thing to do here is to make a custom BC that does this in your example.